File: src/main/java/ee/leola/kassa/models/Feedback.java
package ee.leola.kassa.models;

import jakarta.persistence.Entity;
import jakarta.validation.constraints.NotNull;
import java.time.Instant;
import java.time.format.DateTimeFormatter;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class Feedback extends Model {

  @NotNull private String content;

  private Instant created = Instant.now();

  public String getFormattedTime() {
    return DateTimeFormatter.ISO_INSTANT.format(created);
  }
}

File: src/main/kotlin/ee/leola/kassa/models/Feedback.kt
package ee.leola.kassa.models

import jakarta.persistence.Entity
import jakarta.persistence.Transient
import jakarta.validation.constraints.NotNull
import java.time.Instant
import java.time.format.DateTimeFormatter

@Entity
class Feedback(
    val content: @NotNull String,
    val created: Instant = Instant.now(),
) : Model() {
    @get:Transient
    val formattedTime get() = DateTimeFormatter.ISO_INSTANT.format(created)
}
